addons/purchase/models/purchase.py

```diff
@@ -1216,12 +1216,6 @@ def _product_id_change(self):
             return
 
         self.product_uom = self.product_id.uom_po_id or self.product_id.uom_id
-        product_lang = self.product_id.with_context(
-            lang=get_lang(self.env, self.partner_id.lang).code,
-            partner_id=self.partner_id.id,
-            company_id=self.company_id.id,
-        )
-        self.name = self._get_product_purchase_description(product_lang)
```

Comment:
if I do:

- set productA => name is set to productA
- set productB on same line => name stays productA

so as far as I can tell we can't remove this

I only see two solutions:

- either if we didn't want the two description in the product column => not have 2 description returned for `_get_product_purchase_description`
- or in `_compute_price_unit_and_date_planned_and_name` add annother `default_names.append(line._get_product_purchase_description(line.product_id.with_context(product_ctx)))` but with the context containing partner_id instead of the seller